sound/soc/sof/sof-audio.h

```diff
@@ -332,6 +332,7 @@ struct snd_sof_pcm_stream {
 	struct snd_soc_dapm_widget_list *list; /* list of connected DAPM widgets */
 	bool d0i3_compatible; /* DSP can be in D0I3 when this pcm is opened */
 	unsigned int dsp_max_burst_size_in_ms; /* The maximum size of the host DMA burst in ms */
+	bool chain_dma_allocated; /* indicates the ChainDMA allocation state */
```

Comment:
@plbossart , the `void *private` is used solely for the ipc4 time tracking (delay stuff) in ipc4-pcm.c, and I just moved that struct locally to there. This flag is also used only in ipc4-pcm.c. It will take quite a bit of churn to make space for this flag behind of the `private`....

But unfortunately I agree with you :(